Libraries/PeriphDrivers/Source/I2C/i2c_reva.c

```diff
@@ -245,6 +299,140 @@ int MXC_I2C_RevA_GetClockStretching(mxc_i2c_reva_regs_t *i2c)
     return !((i2c->ctrl & MXC_F_I2C_REVA_CTRL_CLKSTR_DIS) >> MXC_F_I2C_REVA_CTRL_CLKSTR_DIS_POS);
 }
 
+int MXC_I2C_RevA_DMA_Init(mxc_i2c_reva_regs_t *i2c, mxc_dma_reva_regs_t *dma)
+{
+    int8_t i2cNum;
+    int8_t rxChannel;
+    int8_t txChannel;
+    mxc_dma_config_t rxConfig;
```

Comment:
Might be worth considering adding either a parameter or another function to handle TX and RX separately. It seems like there are a fair number of applications which would either want to use DMA for TX or RX, but not necessarily both. If that is the case then it would be kind of silly to take two DMA channels when the user only needs one.